src/System.Memory/src/System/SpanExtensions.cs

```diff
@@ -13,105 +13,102 @@ namespace System
     public static class SpanExtensions
     {
         /// <summary>
-        /// Casts a Span of one primitive type <typeparamref name="T"/> to Span of bytes.
-        /// That type may not contain pointers or references. This is checked at runtime in order to preserve type safety.
+        /// Searches for the specified value and returns the index of its first occurrence. If not found, returns -1. Values are compared using IEquatable&lt;T&gt;.Equals(T). 
         /// </summary>
-        /// <param name="source">The source slice, of type <typeparamref name="T"/>.</param>
-        /// <exception cref="System.ArgumentException">
-        /// Thrown when <typeparamref name="T"/> contains pointers.
-        /// </exception>
-        /// <exception cref="System.OverflowException">
-        /// Thrown if the Length property of the new Span would exceed Int32.MaxValue.
-        /// </exception>
+        /// <param name="span">The span to search.</param>
+        /// <param name="value">The value to search for.</param>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static Span<byte> AsBytes<T>(this Span<T> source)
-            where T : struct
+        public static int IndexOf<T>(this Span<T> span, T value)
+            where T:struct, IEquatable<T>
```

Comment:
Don't we already have this overload in Span? i.e. should it simply be removed from extensions? Same for other indexof overloads.